contracts/dutchX/KyberDutchXReserve.sol

```diff
@@ -0,0 +1,349 @@
+        (destAmount, frtsIssued) = dutchX.claimBuyerFunds(auctionData.dstToken, auctionData.srcToken, this,
+            auctionData.index);
+        dutchX.withdraw(auctionData.dstToken, destAmount);
+
+        if (destToken == ETH_TOKEN_ADDRESS) {
+            weth.withdraw(destAmount);
+            destAmount = destAmount * (BPS - feeBps) / BPS;
+            destAddress.transfer(destAmount);
+        } else {
+            require(auctionData.dstToken.transfer(destAddress, destAmount));
+        }
+
+        require(conversionRate <= calcRateFromQty(
+            srcAmount, /* srcAmount */
+            destAmount, /* destAmount */
+            getDecimals(srcToken), /* srcDecimals */
+            getDecimals(destToken) /* dstDecimals */
+        ));
+        
+        TradeExecute(
+            msg.sender, /* sender */
+            srcToken, /* src */
+            srcAmount, /* srcAmount */
+            destToken, /* destToken */
+            destAmount, /* destAmount */
+            destAddress, /* destAddress */
+            auctionData.index
+        );
+
+        return true;
+    }
+
+    event FeeUpdated(
+        uint bps
+    );
+
+    function setFee(uint bps) public onlyAdmin {
+        require(bps <= BPS);
+        feeBps = bps;
+        FeeUpdated(bps);
+    }
+
+    event TokenListed(
+        ERC20 token
+    );
+
+    function listToken(ERC20 token)
+        public
+        onlyAdmin
+    {
+        require(address(token) != 0);
+
+        listedTokens[token] = true;
+
+        setDecimals(token);
+
+        require(token.approve(dutchX, 2**255));
+
+        TokenListed(token);
+    }
+
+    event TokenDelisted(ERC20 token);
+
+    function delistToken(ERC20 token)
+        public
+        onlyAdmin
+    {
+        require(listedTokens[token] == true);
+        listedTokens[token] == false;
+
+        TokenDelisted(token);
+    }
+
+    event TradeEnabled(
+        bool enable
+    );
+
+    function setDutchXFee() public {
+        (dutchXFeeNum, dutchXFeeDen) = dutchX.getFeeRatio(this);
+
+        // can't use denominator 0 (EVM bad instruction)
+        if (dutchXFeeDen == 0) {
+            tradeEnabled = false;
+        } else {
+            tradeEnabled = true;
+        }
+
+        TradeEnabled(tradeEnabled);
+    }
+
+    function disableTrade()
+        public
+        onlyAlerter
+        returns(bool)
+    {
+        tradeEnabled = false;
+        TradeEnabled(false);
+        return true;
+    }
+
+    event KyberNetworkSet(
+        address kyberNetwork
+    );
+
+    function setKyberNetwork(
+        address _kyberNetwork
+    )
+        public
+        onlyAdmin
+    {
+        require(_kyberNetwork != 0);
+        kyberNetwork = _kyberNetwork;
+        KyberNetworkSet(kyberNetwork);
+    }
+
+    event DutchXSet(
+        DutchXExchange dutchX
+    );
+
+    function setDutchX(
+        DutchXExchange _dutchX
+    )
+        public
+        onlyAdmin
+    {
+        require(_dutchX != DutchXExchange(0));
```

Comment:
will this function really work with all the allowance we need to give? maybe remove it.